internal/controller/git/git.go

```diff
@@ -480,18 +495,56 @@ func (r *repo) WorkingDir() string {
 	return r.dir
 }
 
+func (r *repo) SetAuthor(author CommitUser) error {
+	if author.Name == "" {
+		author.Name = "Kargo Render"
+	}
+
+	cmd := r.buildGitCommand("config", "--global", "user.name", author.Name)
+	cmd.Dir = r.homeDir // Override the cmd.Dir that's set by r.buildGitCommand()
+	if _, err := libExec.Exec(cmd); err != nil {
+		return fmt.Errorf("error configuring git user name: %w", err)
+	}
+
+	if author.Email == "" {
+		author.Email = "kargo-render@akuity.io"
+	}
+
+	cmd = r.buildGitCommand("config", "--global", "user.email", author.Email)
+	cmd.Dir = r.homeDir // Override the cmd.Dir that's set by r.buildGitCommand()
+	if _, err := libExec.Exec(cmd); err != nil {
+		return fmt.Errorf("error configuring git user email: %w", err)
+	}
+
+	if author.SigningKeyPath != "" {
```

Comment:
Git has support for signing commits with [SSH keys](https://docs.gitlab.com/ee/user/project/repository/signed_commits/ssh.html). I think we should take this into account, and should not automatically assume keys to be of a `gpg` type.

Reply:
Let's actually handle ssh key signing in a follow-up and for now, just be sure we don't paint ourselves into a corner that will make that harder later.

Reply:
I went with the first option, defaulting to `gpg`. But all the mechanisms are there to allow us to add SSH signing support in a follow-up.